src/Compilers/CSharp/Portable/Symbols/Source/SourceModuleSymbol.cs

```diff
@@ -501,29 +501,43 @@ internal override void DecodeWellKnownAttribute(ref DecodeWellKnownAttributeArgu
             if (attribute.IsTargetAttribute(this, AttributeDescription.DefaultCharSetAttribute))
             {
                 CharSet charSet = attribute.GetConstructorArgument<CharSet>(0, SpecialType.System_Enum);
-                if (!ModuleWellKnownAttributeData.IsValidCharSet(charSet))
+                if (!CommonModuleWellKnownAttributeData.IsValidCharSet(charSet))
                 {
                     CSharpSyntaxNode attributeArgumentSyntax = attribute.GetAttributeArgumentSyntax(0, arguments.AttributeSyntaxOpt);
                     arguments.Diagnostics.Add(ErrorCode.ERR_InvalidAttributeArgument, attributeArgumentSyntax.Location, arguments.AttributeSyntaxOpt.GetErrorDisplayName());
                 }
                 else
                 {
-                    arguments.GetOrCreateData<ModuleWellKnownAttributeData>().DefaultCharacterSet = charSet;
+                    arguments.GetOrCreateData<CommonModuleWellKnownAttributeData>().DefaultCharacterSet = charSet;
                 }
             }
             else if (attribute.IsTargetAttribute(this, AttributeDescription.NonNullTypesAttribute))
             {
-                bool value = attribute.GetConstructorArgument<bool>(0, SpecialType.System_Boolean);
-                arguments.GetOrCreateData<ModuleWellKnownAttributeData>().NonNullTypes = value;
+                // NonNullTypesAttribute should not be set explicitly.
+                arguments.Diagnostics.Add(ErrorCode.ERR_ExplicitNonNullTypesAttribute, arguments.AttributeSyntaxOpt.Location);
             }
         }
 
         public override bool? NonNullTypes
         {
             get
             {
-               var data = GetDecodedWellKnownAttributeData();
-               return data?.NonNullTypes;
+                string id = "CS" + ((int)ErrorCode.WRN_PragmaNonNullTypes).ToString("0000");
+
+                ReportDiagnostic report;
+                if (_assemblySymbol.DeclaringCompilation.Options.SpecificDiagnosticOptions.TryGetValue(id, out report))
```

Comment:
>report [](start = 111, length = 6)

Nit: Use out param.

Reply:
> Nit: Use out param.

I am assuming you meant "out var" feature. I don't think we have a requirement to use it. Also, I expect this code to go away once we get real compiler switch.

---
In reply to: [224175915](https://github.com/dotnet/roslyn/pull/30387#discussion_r224175915) [](ancestors = 224175915)